app/src/main/java/com/sun/makeupwindow/data/source/local/dao/ProductDAOImpl.kt

```diff
@@ -0,0 +1,71 @@
+package com.sun.makeupwindow.data.source.local.dao
+
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.data.source.local.db.AppDatabase
+
+class ProductDAOImpl private constructor(appDatabase: AppDatabase) :
+    ProductDAO {
+
+    private val appDatabase = appDatabase
+    private val writableDatabase = appDatabase.writableDatabase
+    private val readableDatabase = appDatabase.readableDatabase
+
+    override fun getProduct(): List<Product> {
+        val listProduct = mutableListOf<Product>()
+        val cursor =
+            readableDatabase.query(Product.TABLE_NAME, null, null, null, null, null, null)
+
+        cursor.use {
+            while (cursor.moveToNext()) {
+                val listColor = ColorDAOImpl.getInstance(appDatabase)
+                    .getColor(it.getInt(it.getColumnIndex(Product.ID)))
+                listProduct.add(Product(it, listColor))
+            }
+        }
+        return listProduct
+    }
+
+    override fun getItemProduct(idproduct: Int): Product {
+        val cursur = readableDatabase.query(
+            false, Product.TABLE_NAME, null, Product.ID + "= ?",
+            arrayOf(idproduct.toString()), null, null, null, null
+        )
+        val listColor = ColorDAOImpl.getInstance(appDatabase).getColor(idproduct)
+        return Product(cursur, listColor)
+    }
+
+    override fun getProductByCategory(category: String): List<Product> {
+        val listProduct = mutableListOf<Product>()
+        val query =
+            "SELECT * FROM ${Product.TABLE_NAME} WHERE ${Product.CATEGORY} LIKE '%${category}%'"
+        val cursor =
+            readableDatabase.rawQuery(query,null)
+        cursor.use {
+            while (cursor.moveToNext()) {
+                val listColor = ColorDAOImpl.getInstance(appDatabase)
+                    .getColor(it.getInt(it.getColumnIndex(Product.ID)))
+                listProduct.add(Product(it, listColor))
+            }
+        }
+        return listProduct
+    }
+
+    override fun addProduct(product: Product): Boolean {
+        return writableDatabase.insert(
```

Comment:
return luôn kìa thì dùng = luôn 
```suggestion
    override fun addProduct(product: Product) =
```